### PR TITLE
fix: precompiles are considered COLD when using BESU frame method

### LINKED ISSUE #1067: 2019510-2019519
[ec2-user@ip-10-40-42-125 ~]$  time curl --location --request POST 'localhost:15480' --header 'Content-Type: application/json' --data-raw '{"jsonrpc": "2.0", "method": "linea_generateConflatedTracesToFileV2", "params": [{"startBlockNumber":2019510,"endBlockNumber":2019519 , "expectedTracesEngineVersion": "0.5.1-beta"}], "id": 1}'
{"jsonrpc":"2.0","id":1,"error":{"code":-32603,"message":"Plugin internal error","data":"Exceptions triggered while tracing:\n  - null\n  - Cannot invoke \"net.consensys.linea.zktracer.module.hub.section.TraceSection.fragments()\" because \"section\" is null\n"}}

--

07:45:10.128 [vert.x-worker-thread-12] ERROR net.consensys.linea.zktracer.ZkTracer - First exception that was caught while tracing:
java.lang.IllegalArgumentException: null
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:129) ~[guava-33.0.0-jre.jar:?]
        at net.consensys.linea.zktracer.module.hub.section.call.CallSection.<init>(CallSection.java:153) ~[?:?]
        at net.consensys.linea.zktracer.module.hub.Hub.traceOpcode(Hub.java:1074) ~[?:?]
        at net.consensys.linea.zktracer.module.hub.Hub.processStateExec(Hub.java:958) ~[?:?]
        at net.consensys.linea.zktracer.module.hub.Hub.tracePreExecution(Hub.java:731) ~[?:?]
        at net.consensys.linea.zktracer.ZkTracer.tracePreExecution(ZkTracer.java:258) ~[?:?]
        at org.hyperledger.besu.evm.EVM.runToHalt(EVM.java:207) ~[besu-evm-24.8-develop-915fcb01.jar:24.8-develop-915fcb01]
        at org.hyperledger.besu.evm.processor.AbstractMessageProcessor.codeExecute(AbstractMessageProcessor.java:175) ~[besu-evm-24.8-develop-915fcb01.jar:24.8-develop-915fcb01]
        at org.hyperledger.besu.evm.processor.AbstractMessageProcessor.process(AbstractMessageProcessor.java:198) ~[besu-evm-24.8-develop-915fcb01.jar:24.8-develop-915fcb01]
        at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.process(MainnetTransactionProcessor.java:607) ~[besu-core-24.8-develop-915fcb01.jar:24.8-develop-915fcb01]
        at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.processTransaction(MainnetTransactionProcessor.java:457) ~[besu-core-24.8-develop-915fcb01.jar:24.8-develop-915fcb01]
        at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.processTransaction(MainnetTransactionProcessor.java:229) ~[besu-core-24.8-develop-915fcb01.jar:24.8-develop-915fcb01]
        at org.hyperledger.besu.services.TraceServiceImpl.lambda$trace$6(TraceServiceImpl.java:214) ~[besu-24.8-develop-915fcb01.jar:24.8-develop-915fcb01]
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1597) ~[?:?]
        at org.hyperledger.besu.services.TraceServiceImpl.trace(TraceServiceImpl.java:200) ~[besu-24.8-develop-915fcb01.jar:24.8-develop-915fcb01]
        at org.hyperledger.besu.services.TraceServiceImpl.lambda$trace$2(TraceServiceImpl.java:164) ~[besu-24.8-develop-915fcb01.jar:24.8-develop-915fcb01]
        at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]

### PR DESCRIPTION
**Pull request by letypequividelespoubelles**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/imc/StpCall.java [hunk 1]
```diff
@@ -88,16 +96,15 @@ private void stpCallForCalls(Hub hub) {
           default -> throw new IllegalArgumentException(
               "STP module triggered for a non CALL-type instruction");
         };
-    this.warm = frame.isAddressWarm(to);
+    this.warm = frame.isAddressWarm(to) || isPrecompile(to);
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/imc/StpCall.java]: the only non-formatting change

Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/imc/StpCall.java]: That is a potentially a bug. Did you notice this with the `ToyExecutionEnvironmentV2` or with replay tests @letypequividelespoubelles ?

Comment (letypequividelespoubelles) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/imc/StpCall.java]: with replayTest, but the bug comes from a @FlorianHuc 's block with a BESU node

**Review by OlivierBBB — APPROVED** [review]
The only thing I'm worried about now is in our tracing of account-rows: do we have to manually insert the warmth of precompiles @letypequividelespoubelles ? Given that Besu does not recognize precompiles as inherently warm.

**Comment by letypequividelespoubelles** [comment]
> The only thing I'm worried about now is in our tracing of account-rows: do we have to manually insert the warmth of precompiles @letypequividelespoubelles ? Given that Besu does not recognize precompiles as inherently warm.

Yes, Besu manually add the warmth of precompile when itr deals with calls : see https://github.com/hyperledger/besu/blob/65240fdf4697a81824048d453a0d9818c37c2c3d/evm/src/main/java/org/hyperledger/besu/evm/gascalculator/BerlinGasCalculator.java#L169 (TY @ahamlat ). 

I'll create a method and check that all usage of `isAdressWarm` are ok.

**Pull request merged**